src/Ui/media/Wrapper.coffee

```diff
@@ -280,14 +282,37 @@ class Wrapper
 
 	actionPermissionAdd: (message) ->
 		permission = message.params
+		if Array.isArray(permission) && permission.length > 1
+			@sendInner {"cmd": "response", "to": message.id, "result": {"error": "Can only grant one permission at the time"}}
+			return false
+		if Array.isArray(permission) && permission.length == 1
+			permission = permission[0]
 		$.when(@event_site_info).done =>
 			if permission in @site_info.settings.permissions
+				@notifications.add "notification-#{message.id}", "info", "Permission already granted.", 4000
```

Comment:
Please remove this. This condition was added for the use case when you don't want to check whether you have the permission already.

Reply:
You could pass a single argument instead of an array of arguments but whatever. I think that you should just reply with an "already added" message. (but *not* an error)

Reply:
> You could pass a single argument instead of an array of arguments but whatever.
Yeah, we need to update the documentation though because in the example it is an array (https://zeronet.io/docs/site_development/zeroframe_api_reference/#wrapperpermissionadd). I am not sure how to handle this without breaking compatibility in case people have done the same as in the documentation.

> I think that you should just reply with an "already added" message. (but not an error)
I can remove the notification and replace by a response to the site. However I do have an argument in favor of keeping it. The user is used to be asked if he wants to grant or not a permission. In the case the permission would have already been granted (but the user might not be aware of it) it would not prompt the question and the site might neither inform the user of it. The lack of information and the incoherent experience could be confusing for the user.

Does it make sense to you ?

Reply:
I'm afraid it still doesn't. See, many sites need Merger:whatever permission. They don't check whether they already have it, they just send wrapperPermissionAdd on page load. Currently, it would only bother the user once, but if you add notification it'll show up when the site is loaded and bother the user a lot more.

Reply:
You made a good point. I agree with you regarding the Merger Site permission. I will proceed to modify it.